docs/wiki/EL/data-structures.md

```diff
@@ -126,16 +126,51 @@ The structure `T` consists of the following:
+- **Nonce**: A scalar value identifying the number of transactions successfully sent from this account.
+- **Balance**: The amount of ETH in Wei owned by this account.
+- **Code Hash**: The hash of the EVM code if it's a contract account. For EOAs, it's the keccak-256 hash of an empty string `(keccak256(''))`, which uniquely identifies the account as an EOA.
+- **Storage Root Hash**: The 32 byte hash that points to the root of the account’s ***Storage Trie***, which would be an empty trie for an EOA.
+
+The **World State Trie** is not stored in the chain, but the 32-byte keccak-256 **state root** of the trie is stored in every block header after all transactions in a block have been processed.  The **state root** is used as a cryptographic commitment for the entire system state since it's cryptographically dependent on all the data in the trie.  For example, a node can prove an account's existence given the **state root** and a **Merkle proof** containing the account and it's sibling nodes needed to recreate the **state root**.  Furthermore, the **state root** in each block anchors Ethereum’s consensus: any node can independently compute or verify this root by applying the block’s transactions to the previous state trie.
+
+Below is a simplified diagram of the ***World State Trie***.
 
 ![Merkle Tree](../../images/eth-tries.png)
 
+Let's traverse the trie to find the account with a **45 ETH** balance. The key for this account is shown as `a711355`, meaning these seven hex digits direct us from the root node down to the leaf node.
+
+> This short key `a711355` is just for demonstration. Actual addresses in Ethereum get hashed (32 bytes) and thus typically yield up to 64 nibbles in the trie. But the traversal steps are the same—each nibble selects the next branch/extension node until we arrive at the leaf node storing the final account data.
+
+1. **Key to Nibbles**  
+   - The key string `a711355` represents seven hex digits: `a`, `7`, `1`, `1`, `3`, `5`, `5`.
+   - Each digit is a nibble (4 bits), so the entire key is a sequence of seven nibbles.
+
+2. **Path Through the Trie**  
+   - **Extension node** at the root might store a prefix like `a7`, consuming the first two nibbles.
+   - **Branch node** follows, allowing navigation by each subsequent nibble (`1`, `1`, `3`, `5`, `5`).
+
+3. **Leaf Node**  
+   - Consuming all nibbles brings us to the **leaf node**. In our simplified example, its stored value is **“45 ETH”**.
+   - In Ethereum’s real MPT, this leaf node actually holds the RLP-encoded account object `[nonce, balance, storageRoot, codeHash]`.
+
+### Persistant Storage
+The **World State Trie** is a living structure that evolves with each block, unlike the transaction and receipt tries which are rebuilt from scratch for every block. A full node will keep track of the current state of the **World State Trie** and enough trie nodes required to rewind during a re-org.  Archival nodes will keep track of all previous states since genesis.
```

Comment:
You could reference here that this is why Ethereum is called a state machine. In context of how nodes handle the state, each node needs to know the current state to verify transactions and update the state by executing it. Therefore there is also interim state between transactions but nodes only keep track of the result after the whole block

Reply:
for sure @taxmeifyoucan! updated paragraph to this:
"The **World State Trie** is a living structure that evolves with each block, unlike the transaction and receipt tries which are rebuilt from scratch for every block.  Ethereum operates as a state machine, where the current state is updated by executing transactions in a block. Each node must track this current state to verify transactions and update it accordingly. Therefore, intermediate states exist during block processing, but nodes retain only the final post-block state.  A full node will keep track of the current state of the **World State Trie** and enough trie nodes required to rewind during a re-org.  Archival nodes will keep track of all previous states since genesis."